### ISSUE TITLE
Help generating a blind payload for Apache Commons Collections 3 Alternate Payload 2 

### ISSUE THREAD

**Issue by nchigmaroff** [federicodotta/Java-Deserialization-Scanner#34]
Hi Federico, 

First of all, thank you for your amazing scanner! It detected a vulnerability in my pentesting target using the Apache Commons Collections 3 Alternate Payload 2 (and 3) that I certainly would not have found otherwise. I was able to replicate its payload myself by modifying ysoserial's CommonsCollections5.java file based on the code snippet you included in [#10 ](https://github.com/federicodotta/Java-Deserialization-Scanner/issues/10).

Unfortunately for me, my target appears to be firewalled and I have been unsuccessful in pinging my host or doing DNS lookups. For this reason, I am hoping to modify the gadget chain in the link above to sleep conditionally, as one might do with blind SQLi. For example, the payload would be of the following structure: 

(read current directory name)
(if current directory name starts with "a", sleep 5 seconds)

or more generally

(execute commands to establish if a certain condition is true/false)
(sleep if true)

as opposed to just 

(sleep 5 seconds). 

I have been unsuccessful in doing this, despite a number of different attempts. I have tried to change one of the InvokerTransformers to a SwitchTransformer (in hopes of defining an appropriate predicate), but regardless of what I change ysoserial throws me a serialization error when I attempt to use my compiled .jar file. 

I have also looked at the [yolosec blind fork](https://github.com/yolosec/ysoserial/tree/master) of ysoserial, which includes functionality for such cases. However, my Java knowledge is insufficient to understand the gadget chaining they are doing to the level where I can modify it for my own purposes. 

For this reason, I am creating this issue here to ask if you have any advice or sample payloads for creating the above gadget chain using the structure of #10 . I recognize blind exfiltration is outside the scope of this scanner, but regardless I would appreciate any input on this issue. Hopefully it can also help others who find a similar issue in their testing. 

Thank you for your time!

**Comment by nobodynate** [comment]
@nchigmaroff I was having a similar issue while studying for a popular webapp pentesting course and I also considered the possibility of there being a firewall in the way. I ended up generating a payload to `curl <collaboratorurl>` for each library available in YSOSerial and using intruder to figure out which one works. 

So in my case, the scanner detected CommonsCollections3 alt payloads 3 and 4 but this did not work for manual exploitation. I had to use CommonsCollections6 and everything went fine.
